src/APIRequestWrapper.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Eppo;
+
+use Eppo\Exception\HttpRequestException;
+use Http\Discovery\Psr18Client;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Teapot\StatusCode\RFC\RFC7231;
+use Webclient\Extension\Redirect\RedirectClientDecorator;
+
+class APIRequestWrapper
```

Comment:
@typotter could you please add a comment why this class is needed.
This implementation differs from all other SDKs we have, and new devs could be confused

Reply:
This file started out as a straight copy from HttpClient.php but that history is lost by git.
I've clarified the purpose of this class in a comment. The other SDKs are using the moniker "HttpClient" for this logic, but I found that to be confusing in the context of migrating to use the PSR18 HttpClient.